.github/workflows/dependencies-head.yml

```diff
@@ -140,12 +144,13 @@ jobs:
 
     - name: Install dependencies
       run: |
-        python -m pip install --upgrade pip setuptools wheel
-        python -m pip --no-cache-dir --quiet install --upgrade ".[all,test]"
-        python -m pip uninstall --yes matplotlib
+        python -m pip install uv
+        uv pip install --system --upgrade pip setuptools wheel
+        uv pip --no-cache --quiet install --system --upgrade ".[all,test]"
+        uv pip uninstall --yes matplotlib
```

Comment:
Note to self, see if this (not using `--system`) causes a problem by running workflow